src/quo2/components/loaders/skeleton_list/view.cljs

```diff
@@ -0,0 +1,78 @@
+(ns quo2.components.loaders.skeleton-list.view
+  (:require [quo2.theme :as theme]
+            [react-native.core :as rn]
+            [quo2.foundations.colors :as colors]
+            [react-native.masked-view :as masked-view]
+            [react-native.reanimated :as reanimated]
+            [quo2.components.loaders.skeleton-list.style :as style]
+            [reagent.core :as reagent]
+            [quo2.components.loaders.skeleton-list.constants :as constants]))
+
+(defn skeleton-item
+  [{:keys [index parent-index content color animated?]}]
+  [:f>
+   (fn []
+     (let [loading-color           (colors/theme-colors colors/neutral-10 colors/neutral-60)
+           {window-width :width}   (rn/get-window)
+           translate-x             (reanimated/use-shared-value (- window-width))
+           animated-gradient-style (reanimated/apply-animations-to-style
+                                    {:transform [{:translateX translate-x}]}
+                                    {:width  window-width
+                                     :height "100%"})]
+
+       (reanimated/animate-shared-value-with-repeat translate-x window-width 1000 :linear (- 1) false)
+       [masked-view/masked-view
+        {:style               {:height (get-in constants/layout-dimensions [content :height])}
```

Comment:
Hi @FFFra,
Thank you very much for your awesome work.
This implementation starting to look like the old implementation we had.
We were avoiding using the old implementation because masked view & animations are heavy and also unnecessary for the static skeleton.
And they were slowing down, the opening of the chat screen. (As shown in the https://github.com/status-im/status-mobile/pull/16438#issuecomment-1618185714)

Please can we create two components, a `static-skeleton-item` and an `animated-skeleton-item` component? Where animated version derives from the static one (Probably static can be used as a `maskElement` prop).

And in the `internal-view` function, we can use one of them based on the `animated?` condition. This way static version will be much simpler and light. wdyt?

Reply:
(thinking out loud)
Instead of using masked view in every animated skeleton item, can we only have a single masked view for complete skeleton view. This will probably also increase animated version performance.

Reply:
I have something in mind for that. I would like to hear your opinion. Let me formulate and I will ping you on Discord